src/app/(Authentication)/register/page.tsx

```diff
@@ -0,0 +1,194 @@
+'use client';
+import React from 'react';
+import _ from 'lodash';
+import { Button, ButtonStyle, Input } from '@/components/formElements';
+import { FormEvent, useState } from 'react';
+import Link from 'next/link';
+import { RegistrationFields } from '@/utils/FormFields';
+import { regExPatterns, testPassword } from '@/utils';
+import swal from 'sweetalert';
+import { useRouter } from 'next/navigation';
+import { useSelector } from 'react-redux';
+import { registerUser } from '@/redux/slices/RegisterSlice';
+import { RootState } from '@/redux/store';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks/hook';
+import useToast from '@/components/alerts/Alerts';
+import { ToastContainer } from 'react-toastify';
+
+export interface FormDataInterface {
+  name: string;
+  email: string;
+  password: string;
+  phone: string;
+  address: string;
+}
+
+export interface ErrorInterface {
+  target: RegistrationKeys;
+  msg: string;
+}
+
+export type RegistrationKeys = keyof FormDataInterface;
+
+const InitialFormValues: FormDataInterface = {
+  name: '',
+  email: '',
+  password: '',
+  phone: '',
+  address: ''
+};
+
+export interface RegisterError {
+  status: 'Success!' | 'Error';
+  message: string;
+  data: { field: string; message: string }[];
+}
```

Comment:
can you declare this type outside because this will be used everytime

Reply:
Yes, I have added them in utils to be reused, but is will be like this only for forms that's why I renamed it to **FormErrorInterface** 👍